diff --git a/CMakeLists.txt b/CMakeLists.txt
index 953e6ad..8191bfe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -146,12 +146,8 @@ else()
 endif()
 
 if(BUILD_COMMON_CURL)
-  if (WIN32)
-    find_package(CURL REQUIRED)
-    set(CURL_LIBRARIES CURL::libcurl)
-  else()
-    pkg_check_modules(CURL REQUIRED libcurl)
-  endif()
+  find_package(CURL REQUIRED)
+  set(CURL_LIBRARIES CURL::libcurl)
 
   set(OPEN_SRC_INCLUDE_DIRS ${OPEN_SRC_INCLUDE_DIRS} ${CURL_INCLUDE_DIRS})
   link_directories(${CURL_LIBRARY_DIRS})
@@ -281,15 +277,6 @@ if(BUILD_COMMON_CURL)
       "${CMAKE_CURRENT_SOURCE_DIR}/libkvsCommonCurl.pc.cmake"
       "${CMAKE_CURRENT_BINARY_DIR}/libkvsCommonCurl.pc" @ONLY)
 
-    if(BUILD_STATIC)
-      # Curl will enable ZLIB as part of its build if it finds
-      # the package. We need to therefore link it for static builds. 
-      find_package(ZLIB)
-      if(ZLIB_FOUND)
-        list(APPEND CURL_LIBRARIES z)
-      endif()
-    endif()
-
     add_library(kvsCommonCurl STATIC ${KVS_COMMON_SOURCE_FILES_BASE} ${KVS_COMMON_SOURCE_FILES_CURL})
     target_compile_definitions(kvsCommonCurl PRIVATE KVS_BUILD_WITH_CURL ${CPRODUCER_COMMON_TLS_OPTION})
     target_link_libraries(kvsCommonCurl
